pallets/amm/src/lib.rs

```diff
@@ -389,70 +399,78 @@ decl_module! {
 				Error::<T>::InsufficientPoolAssetBalance
 			);
 
-			// Note: this check is not really needed as we already check if amount to remove >= liquidity in pool
 			let liquidity_left = total_shares.checked_sub(liquidity_amount).ok_or(Error::<T>::InvalidLiquidityAmount)?;
 
-			T::Currency::transfer(asset_a, &pair_account, &who, remove_amount_a)?;
-			T::Currency::transfer(asset_b, &pair_account, &who, remove_amount_b)?;
+			with_transaction_result( || {
+				T::Currency::transfer(asset_a, &pair_account, &who, remove_amount_a)?;
```

Comment:
Is this needed? We check everything beforehand. I feel this is either more computation or the ensures are now not needed? What does this add? Is the performance same? Maybe I need to check more how it works?

Reply:
True. we check everything. However if for some reason, something happens, let's say in the second transfer ( albeit unlikely ) - we are safe.  This was suggested in the builder's review.  Not sure if it has performance impact - i will check but imo, it is none or insignificant.

Reply:
I'll leave this in as i believe it does not have any performance impact. Added a question to the list for builders 5h ama.